src/utils/graph_building.jl

```diff
@@ -36,45 +38,56 @@ function build_graph(
     cutoff_radius::Real = 8.0,
     max_num_nbr::Integer = 12,
     dist_decay_func::Function = inverse_square,
-    normalize_weights::Bool = true,
 )
-    aseio = pyimport_conda("ase.io", "ase", "conda-forge")
-    atoms_object = aseio.read(file_path)
-
-    # list of atom symbols
-    atom_ids = [get(atoms_object, i - 1).symbol for i = 1:length(atoms_object)]
-
-    # check if any nonperiodic BC's
-    nonpbc = any(.!atoms_object.pbc)
-    local cant_voronoi = false
-    if nonpbc & use_voronoi
-        @warn "Voronoi edge weights are not supported if any direction in the structure is nonperiodic. Using cutoff weights method..."
-        cant_voronoi = true
-    end
+    c = Crystal(file_path)
+    atom_ids = String.(c.atoms.species)
 
-    if use_voronoi && !cant_voronoi
+    if use_voronoi
+        @info "Note that building neighbor lists and edge weights via the Voronoi method requires the assumption of periodic boundaries. If you are building a graph for a molecule, you probably do not want this..."
         s = pyimport_conda("pymatgen.core.structure", "pymatgen", "conda-forge")
-        pmgase = pyimport_conda("pymatgen.io.ase", "pymatgen", "conda-forge")
-        aa = pmgase.AseAtomsAdaptor()
-        struc = aa.get_structure(atoms_object)
+        struc = s.Structure.from_file(file_path)
         weight_mat = weights_voronoi(struc)
+        return weight_mat, atom_ids
     else
-        nl = pyimport_conda("ase.neighborlist", "ase", "conda-forge")
-        is, js, dists = nl.neighbor_list("ijd", atoms_object, cutoff_radius)
-        weight_mat = weights_cutoff(
-            is .+ 1,
-            js .+ 1,
-            dists;
+        build_graph(
+            c;
+            cutoff_radius = cutoff_radius,
             max_num_nbr = max_num_nbr,
             dist_decay_func = dist_decay_func,
         )
     end
 
-    if normalize_weights
-        weight_mat = weight_mat ./ maximum(weight_mat)
-    end
+end
+
+"""
+Build graph from a Crystal object. Currently only supports the "cutoff" method of neighbor list/weight calculation (not Voronoi).
+This dispatch exists to support autodiff of graph-building.
+
+# Arguments
+## Required Arguments
+- `crys::Crystal`: Crystal object representing the atomic geometry from which to build a graph
+
+## Keyword Arguments
+- `cutoff_radius::Real=8.0`: cutoff radius for atoms to be considered neighbors (in angstroms)
+- `max_num_nbr::Integer=12`: maximum number of neighbors to include (even if more fall within cutoff radius)
+- `dist_decay_func::Function=inverse_square`: function to determine falloff of graph edge weights with neighbor distance
+"""
+function build_graph(
```

Comment:
Could we have a reference to this docstring also under `Utilities/Graph Building`?

Reply:
shouldn't the `@autodocs` do that?

Reply:
I can't check right now because my local setup for building the docs is somehow broken again :/ but I thought the `@autodocs` block should do it?